### PR TITLE
Update EVP_DigestSignInit() docs

### LINKED ISSUE #5854: EVP_DigestSignInit fails with BLAKE2 algorithms
The `EVP_DigestSignInit` function fails when using the `EVP_blake2s256` and `EVP_blake2b512` algortihms. I'm not sure if this is by design or if those algorithms are not supported. I assumed any of the `EVP_MD` types could be used and I couldn't find anything in the docs about supported types. I'm using the 1.1.0f version of the libcrypto library.

### PR DESCRIPTION
**Pull request by mattcaswell**
Explicitly state which digests can be used with which algorithms.

Fixes #5854

### CODE REVIEW

**Comment by mattcaswell** [comment]
It's not clear to me if the restriction that you cannot use SHA3 with ECDSA is intentional or not. Perhaps @dot-asm can comment.

File: doc/man3/EVP_DigestSignInit.pod [hunk 1]
```diff
@@ -26,7 +26,25 @@ EVP_DigestSignInit() sets up signing context B<ctx> to use digest B<type> from
 ENGINE B<e> and private key B<pkey>. B<ctx> must be created with
 EVP_MD_CTX_new() before calling this function. If B<pctx> is not NULL the
 EVP_PKEY_CTX of the signing operation will be written to B<*pctx>: this can
-be used to set alternative signing options.
+be used to set alternative signing options. The digest B<type> may be NULL if
+the signing algorithm supports it.
+
+Only EVP_PKEY types that support signing can be used with these functions. This
+includes MAC algorithms where the MAC generation is considered as a form of
+"signing". Built-in EVP_PKEY types supported by these functions are CMAC,
```
Comment (richsalz) [inline doc/man3/EVP_DigestSignInit.pod]: Punctuation inside the quotes.

File: doc/man3/EVP_DigestSignInit.pod [hunk 2]
```diff
@@ -26,7 +26,25 @@ EVP_DigestSignInit() sets up signing context B<ctx> to use digest B<type> from
...
+Poly1305, DSA, HMAC, RSA, Siphash, Ed25519 and Ed448.
+
+Not all digests can be used for all key types. The following combinations apply.
+DSA supports SHA1, SHA224, SHA256, SHA384 and SHA512. ECDSA supports SHA1,
```
Comment (richsalz) [inline doc/man3/EVP_DigestSignInit.pod]: Would this be more digestible as a table or list?  (E.g., using over/item/back constructs). At a minimum, it should be alphabetized.

**Comment by richsalz** [comment]
Look at https://tools.ietf.org/html/draft-turner-lamps-adding-sha3-to-pkix-01  It's clearly in the works, so if we can support it we should probably do so.

File: doc/man3/EVP_DigestSignInit.pod (hunk 1 above)
Comment (mattcaswell) [reply doc/man3/EVP_DigestSignInit.pod]: Done

File: doc/man3/EVP_DigestSignInit.pod (hunk 2 above)
Comment (mattcaswell) [reply doc/man3/EVP_DigestSignInit.pod]: Ok, done.

**Comment by mattcaswell** [comment]
Fixup commit pushed addressing comments.

**Comment by levitte** [comment]
Any thoughts on backporting this to 1.1.0?

File: doc/man3/EVP_DigestSignInit.pod [hunk 3]
```diff
@@ -26,7 +26,54 @@ EVP_DigestSignInit() sets up signing context B<ctx> to use digest B<type> from
...
 EVP_MD_CTX_new() before calling this function. If B<pctx> is not NULL the
 EVP_PKEY_CTX of the signing operation will be written to B<*pctx>: this can
-be used to set alternative signing options.
+be used to set alternative signing options. The digest B<type> may be NULL if
+the signing algorithm supports it.
+
+Only EVP_PKEY types that support signing can be used with these functions. This
+includes MAC algorithms where the MAC generation is considered as a form of
+"signing." Built-in EVP_PKEY types supported by these functions are CMAC,
+Poly1305, DSA, HMAC, RSA, Siphash, Ed25519 and Ed448.
```
Comment (FdaSilvaYY) [inline doc/man3/EVP_DigestSignInit.pod]: SipHash ?

**Comment by mattcaswell** [comment]
Pushed (including @FdaSilvaYY typo fix). PR coming up for 1.1.0

**Pull request closed without merging**